examples/lfric/eg3/solver_mod.x90

```diff
@@ -503,7 +504,7 @@ contains
 
     ! The scalars
     real(kind=r_def)                   :: h(GCRK+1, GCRK), u(GCRK), g(GCRK+1)
-    real(kind=r_def)                   :: beta, si, ci, nrm, h1, h2, p, q
+    real(kind=r_def)                   :: beta, si, ci, nrm, h1, h2, px, q
```

Comment:
What's the reason for this change?

Reply:
We don't support a variable with the same name being declared as a different type within a file. In this file `p` is declared as something else, so I renamed it.

Reply:
Yes, I agree it is not ideal. However, it does not affect any of the current LFRic algorithm code, other than finding problems with them not declaring scalar precision in a few places - which is an error on their part. You are right that once we have PSyIR working we will then this restriction will be removed as we will support scoping. I certainly do not want to add in scoping to this temporary implementation.